tapalcatl_server/handler.go

```diff
@@ -149,19 +154,126 @@ func (logState *logEntryState) String() string {
 	return result
 }
 
-func MetatileHandler(p Parser, metatileSize int, mimeMap map[string]string, storage Storage, bufferManager BufferManager, logger *log.Logger) http.Handler {
+type metricsWriter interface {
+	Write(*requestState)
+}
+
+type nilMetricsWriter struct{}
+
+func (_ *nilMetricsWriter) Write(reqState *requestState) {}
+
+type statsdMetricsWriter struct {
+	respState  [ResponseState_Count]metrics.Counter
+	fetchState [FetchState_Count]metrics.Counter
+
+	fetchSizeBody metrics.Gauge
+	fetchSizeLen  metrics.Gauge
+	fetchSizeCap  metrics.Gauge
+
+	lastMod metrics.Counter
+	etag    metrics.Counter
+
+	zipErr       metrics.Counter
+	respWriteErr metrics.Counter
+	condErr      metrics.Counter
+}
+
+func (smw *statsdMetricsWriter) Write(reqState *requestState) {
+	respStateInt := int(reqState.responseState)
```

Comment:
Nitpick: We've aliased `reqResponseState` to `int32`, so we really ought to use the same type here.